### ISSUE TITLE
[BUG] Getting Val_AUC metric as 0 along with other losses like regularization_loss etc

### ISSUE THREAD

**Issue by mamtaed8811** [NVIDIA-Merlin/models#284]
### Bug description

I am following along the notebook 2b that was used for the GTC dry run on my own dataset. When I run model.fit, it trains but gives me "val_auc: 0.0000e+00" on every model that I have tried. I would like to understand why this is happening and I am missing anything here. Also, is this auc metric AUC curve or accuracy of a model? I would like comprehend other metrics displayed in the progress bar of these models as well like in the below snapshot, if auc is accuracy, then the training accuracy being 50%, how is the loss 68% as these numbers don't add up to 100%. Similarly, val_auc is 0 but val_loss is .8095. 

I am attaching sample training data and test data parquet files as well as the code snapshots of this training.

### Steps/Code to reproduce bug

1.  import os
    os.environ["TF_GPU_ALLOCATOR"]="cuda_malloc_async"
    import cudf
    import glob
    import gc
    import pandas as pd
    
    import nvtabular as nvt
    from nvtabular.ops import *
    from fit_transform import workflow_fit_transform
    
    from merlin.schema.tags import Tags
    from merlin.schema import Schema
    
    import merlin.models.tf as mm
    import merlin.models.tf.dataset as tf_dataloader
    
    from merlin.io.dataset import Dataset
    from merlin.schema.io.tensorflow_metadata import TensorflowMetadata
    from merlin.models.tf.blocks.core.aggregation import CosineSimilarity
    
    import tensorflow as tf

    train_path = '/workspace/data/train/*.parquet'
    test_path = '/workspace/data/test/*.parquet'
    output_path = '/workspace/data/processed'

2.   user_id = ["userid"] >> AddMetadata(tags=[Tags.USER_ID]) >> Categorify()
      item_id = ["contentid"] >> AddMetadata(tags=[Tags.ITEM_ID]) >> Categorify()
      targets = ["relevant"] >> AddMetadata(
          tags=[str(Tags.BINARY_CLASSIFICATION), "target"]
      )
      
      outputs = user_id+item_id+targets
      
      etl_description = 'etl1'

      workflow_fit_transform(outputs, train_path, test_path, output_path)
      schema = TensorflowMetadata.from_proto_text_file(output_path + '/train/').to_merlin_schema()
      target_column = schema.select_by_tag(Tags.TARGET).column_names[0]
      schema.column_names
      
      model = mm.MatrixFactorizationModel(
          schema,
          dim=64,
          prediction_tasks=mm.BinaryClassificationTask(target_column, metrics=[tf.keras.metrics.AUC()]), 
          aggregation=CosineSimilarity()
      )
      
      model_description = 'MF'
      
      model.compile('SGD', run_eagerly=False)
      
      batch_size = 16*1024
      train_dl = tf_dataloader.BatchedDataset(
          Dataset(output_path + '/train/*.parquet', part_size="500MB"),
          batch_size = batch_size,
          label_names = [target_column],
          shuffle= True,
          schema = schema,
      )
      
      test_dl = tf_dataloader.BatchedDataset(
          Dataset(output_path + '/test/*.parquet', part_size="500MB"),
          batch_size = batch_size,
          label_names = [target_column],
          shuffle = False,
          schema = schema,
      )

3.  model.fit(train_dl, validation_data=test_dl)

### Environment details
- Merlin version: 22.03
- Platform: Ubuntu 18.04
- Python version: 3.8
- PyTorch version (GPU?):
- Tensorflow version (GPU?): 2.8

### Additional context

Dataset:
userid: unique ids of user
contentid: unique ids of items
relevant: target column (whether a user interacted with the item or not, interactions here are viewing a page, updating a page or creating the page)
I am attaching a code snippet of how I am generating this binary column.

I have used other user features as well with merlin models ie MLP, MF and DLRM but I still get val_auc as 0 with all of them.
![Screen Shot 2022-03-21 at 10 39 42 AM](https://user-images.githubusercontent.com/28987826/159348987-2733ba42-9d61-4041-8c20-63c5e2e83ee6.png)
[test_data.csv](https://github.com/NVIDIA-Merlin/models/files/8318740/test_data.csv)
[train_data.csv](https://github.com/NVIDIA-Merlin/models/files/8318743/train_data.csv)
![Screen Shot 2022-03-21 at 12 35 59 PM](https://user-images.githubusercontent.com/28987826/159350463-cb13abd8-2f25-4400-9ae3-ee644f8a711c.png)

**Comment by rnyak** [comment]
@mamtaed8811 remove `metrics=[tf.keras.metrics.AUC()]`  from  the `mm.BinaryClassificationTask(target_column, metrics=[tf.keras.metrics.AUC()])`  then you will get other metrics printed out.

**Comment by mamtaed8811** [comment]
Thanks @rnyak. I am able to print other metrics now and comprehend it a bit better.

**Comment by viswa-nvidia** [comment]
@rnyak Can this be closed ?

**Comment by mamtaed8811** [comment]
@viswa-nvidia and @rnyak . I am seeing that the performance metrics for the trained models like accuracy and loss don't add up any time in my training. It's a bit confusing as to why it doesn't so I wanted help understanding why it isn't.

**Comment by netinupur** [comment]
> Thanks @rnyak. I am able to print other metrics now and comprehend it a bit better.

what was the issue here, I am having the same issue and unable to understand what is happening

**Comment by rnyak** [comment]
@netinupur can you please provide more details?what's your env?   which model you are running?  and what error you get and what's your model build and training script? thanks.
